packages/website/src/components/OptionsSelector.tsx

```diff
@@ -0,0 +1,188 @@
+        config={state.tsConfig}
+        onClose={updateTsConfig}
+      />
+      <Expander label="Info">
+        <label className={styles.optionLabel}>
+          TypeScript
+          <Dropdown
+            name="ts"
+            className="text--right"
+            value={state.ts}
+            onChange={updateTS}
+            options={((tsVersions.length && tsVersions) || [state.ts]).filter(
+              item => parseFloat(item) >= 3.3,
+            )}
+          />
+        </label>
+        <label className={styles.optionLabel}>
+          Eslint
+          <span>{process.env.ESLINT_VERSION}</span>
+        </label>
+        <label className={styles.optionLabel}>
+          TSEslint
+          <span>{process.env.TS_ESLINT_VERSION}</span>
+        </label>
+      </Expander>
+      <Expander label="Options">
+        <label className={styles.optionLabel}>
+          Enable jsx
+          <Checkbox
+            name="jsx"
+            checked={state.jsx}
+            onChange={(e): void => setState({ jsx: e })}
+            className={styles.optionCheckbox}
+          />
+        </label>
+        <label className={styles.optionLabel}>
+          Show AST
+          <Checkbox
+            name="ast"
+            checked={state.showAST}
+            onChange={(e): void => setState({ showAST: e })}
+            className={styles.optionCheckbox}
+          />
+        </label>
+        <label className={styles.optionLabel}>
+          Source type
+          <Dropdown
+            name="sourceType"
+            value={state.sourceType}
+            onChange={(e): void => setState({ sourceType: e as SourceType })}
+            options={['script', 'module']}
+          />
+        </label>
+        <label
+          className={styles.optionLabel}
+          onClick={(): void => setEslintModal(true)}
+        >
+          Eslint Config
+          <EditIcon className={styles.clickableIcon} />
+        </label>
+        <label
+          className={styles.optionLabel}
+          onClick={(): void => setTypeScriptModal(true)}
+        >
+          TypeScript Config
+          <EditIcon className={styles.clickableIcon} />
+        </label>
+      </Expander>
+      <Expander label="Actions">
+        <label className={styles.optionLabel} onClick={copyLinkToClipboard}>
+          Copy Link
+          <Tooltip open={copyLink} text="Copied" close={setCopyLink}>
+            <CopyIcon className={styles.clickableIcon} />
+          </Tooltip>
+        </label>
+        <label className={styles.optionLabel} onClick={copyMarkdownToClipboard}>
+          Copy Markdown
+          <Tooltip open={copyMarkdown} text="Copied" close={setCopyMarkdown}>
+            <CopyIcon className={styles.clickableIcon} />
+          </Tooltip>
+        </label>
+        <label className={styles.optionLabel} onClick={openIssue}>
+          Report Issue
+          <CopyIcon className={styles.clickableIcon} />
+        </label>
+      </Expander>
+    </>
+  );
+}
```

Comment:
Very non-blocking ideation: I wonder if we could use a schema/factory approach to automate creating this side... it'd probably be more code and probably not worth it 🤷

```ts
const sections = [
  {
    heading: "Info",
    inputs: [
      {
        label: "TypeScript",
        options: [ /* ... */ ]
        type: "select",
      },
      // ...
    ],
  },
  // ...
];
```

```ts
sections.map(section => <SidebarSection {...section} key={section.heading} />);
```

Reply:
let's follow up on this with a future PR -  this is already v big.